### DISCUSSION TITLE
Mouse cursor does not move

### DISCUSSION THREAD

**Discussion by mbeltoft** [debauchee/barrier#1792] [Q&A]
I've just updated to Ubuntu 22.04, and then reinstalled barrier to get the latest version (GUI says 2.4.0-release-00000000).
I've configured it, and I can see from the log that I get the log entries:
INFO: entering screen
INFO: leaving screen
But the mouse cursor does not move, and clicking with the mouse does nothing.
What am I doing wrong?

**Comment by midenok** [comment]
# Barrier Mouse Stuck in Bottom Right Corner on Client

When using Barrier, if your mouse cursor enters the client but gets stuck in the bottom right corner and won't move, it's usually due to display scaling or resolution mismatches between server and client.

## Problem

- Mouse enters client (client says "entering screen")
- Cursor is stuck in bottom right corner, cannot move
- Only way to return is to move mouse right, back to host

## Cause

Barrier often miscalculates screen boundaries when:
- Server or client uses display scaling above 100% (common with HiDPI/4K monitors)
- There are mismatched resolutions or multi-monitor setups
- Client is running Linux with Wayland instead of Xorg

## Solutions

| Solution                        | Where to Apply   | Effectiveness   |
|----------------------------------|------------------|-----------------|
| Set scaling to 100%              | Server           | High            |
| Downgrade to Barrier 2.3.4       | Server           | High            |
| Use single monitor               | Server           | Moderate        |
| Switch to Xorg (not Wayland)     | Linux client     | High (if Linux) |

### Steps to Fix

1. **Set Display Scaling to 100%**
   - On Windows:  
     `Settings → System → Display → Scale and layout → 100%`
   - Do this for all monitors on the server.
   - Restart both Barrier server and client.

2. **Downgrade Barrier**
   - Some users report that version 2.3.4 works better than newer versions.
   - Try installing Barrier 2.3.4 on the server.

3. **Simplify Monitor Setup**
   - Disconnect extra monitors from the server.
   - Match resolutions and scaling on both server and client if possible.

4. **Check Display Server (Linux Only)**
   - If your client runs Linux, ensure it uses Xorg, not Wayland.

## Additional Tips

- Double-check that the client screen name in the server config matches exactly (case-sensitive).
- Make sure both computers are on the same network and show as connected in Barrier.
- Restart both server and client after making changes.

---

If these steps don't resolve the issue, provide your OS versions, monitor setup, and Barrier versions for more targeted help.

**Reply by midenok** [reply]
#2024
#1918
https://github.com/debauchee/barrier/issues/1918#issuecomment-1714507813
#2042
...

Looks like nobody watches for duplicates...

**Reply by midenok** [reply]
I confirm 2.3.4 works

**Reply by nbolton** [reply]
Please try Deskflow or Input Leap as Barrier is no longer in development.

https://github.com/deskflow/deskflow
https://github.com/input-leap/input-leap

Both are compatible with Barrier client/server so you can try on one of your computers.

If this is still an issue in those projects, we would appreciate a cross-post of this issue.

Not sure which to use? The 'Insights' tab of each project may help you decide.
